app/javascript/src/Stats/buyer/stats_application.jsx

```diff
@@ -7,6 +7,13 @@ import {StatsMethodsTable} from 'Stats/lib/methods_table'
 import {StatsApplicationMetricsSource} from 'Stats/lib/application_metrics_source'
 import {StatsCSVLink} from 'Stats/lib/csv_link'
 import {Stats} from 'Stats/lib/stats'
+import {StatsStore} from 'Stats/lib/store'
+import $ from 'jquery'
+
+function getStoredApplicationId () {
+  let storedState = new StatsStore(window).getStateFromURL()
+  if (storedState && storedState.selectedApplicationId) return storedState.selectedApplicationId
```

Comment:
`return storedState && storedState.selectedApplicationId`

Reply:
that will return a boolean, not the selected application id

Reply:
No, it will return `number | undefined`. Javascript logic operators do not parse to boolean, only `!` does.

Reply:
Tested in the Chrome Developer console, it doesn't cast to boolean 👍 
![image](https://user-images.githubusercontent.com/11318903/52265102-9c11e480-2933-11e9-8fd2-9f7407337c6e.png)

Reply:
true, mea culpa.

Reply:
> Tested in the Chrome Developer console, it doesn't cast to boolean 👍

Or also add a spec